src/xunit.v3.core/Sdk/v3/Runners/TestInvoker.cs

```diff
@@ -48,7 +48,24 @@ public abstract class TestInvoker<TContext>
 	{
 		Guard.ArgumentNotNull(ctxt);
 
-		return ctxt.TestMethod.Invoke(testClassInstance, ctxt.TestMethodArguments);
+		if (TestEventSource.Log.IsEnabled())
```

Comment:
How expensive/dynamic is this (meaning, is this worth caching)? I assume it's cheap since it would be counterproductive for it to be expensive, but I really don't know anything about the event source pipeline.

Reply:
Or even, does consistency matter? (i.e., does it matter if there's a stop without first seeing a start?)

Reply:
> How expensive/dynamic is this (meaning, is this worth caching)?

It's very cheap (it was designed to have absolute minimal overhead), we don't even cache it in BCL: https://github.com/search?q=repo%3Adotnet%2Fruntime%20%22Log.IsEnabled%22&type=code

> does it matter if there's a stop without first seeing a start?

this is very important, otherwise the Concurrency Visualizer complains and can not visualize it properly

Reply:
Does that mean we should only check once at the start, and only send the stop if we sent the start? That's a pretty simple change.